packages/@aws-cdk/aws-stepfunctions-tasks/lib/lambda/invoke.ts

```diff
@@ -0,0 +1,149 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core';
+import { getResourceArn, TaskStateConfig, taskStateJson, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * Properties for invoking a Lambda function with LambdaInvoke
+ */
+export interface LambdaInvokeProps extends sfn.TaskStateBaseProps {
+
+  /**
+   * Lambda function to invoke
+   */
+  readonly lambdaFunction: lambda.IFunction;
+
+  /**
+   * The JSON that will be supplied as input to the Lambda function
+   *
+   * @default - The state input (JSON path '$')
+   */
+  readonly payload?: sfn.TaskInput;
+
+  /**
+   * Invocation type of the Lambda function
+   *
+   * @default InvocationType.REQUEST_RESPONSE
+   */
+  readonly invocationType?: LambdaInvocationType;
+
+  /**
+   * Client context to pass to the function
+   *
+   * @default - No context
+   */
+  readonly clientContext?: string;
+
+  /**
+   * Version or alias of the function to be invoked
+   *
+   * @default - No qualifier
+   */
+  readonly qualifier?: string;
```

Comment:
This should come from the `IFunction`, methinks. But if we're unsure, I guess we just pass it?

Reply:
had a similar thought when I was copying over the current implementation. Despite taking a closer look at it's usage, I'm still unsure. I'm okay with passing it though as I don't see it being problematic

Reply:
I'm not sure, let me find out.

Reply:
Actually, this maybe useful if the qualifier is expected to be dynamic (eg, using aliases).

I'm ok with leaving this here.

Reply:
@nija-at  sounds good!  I was just in the middle of finding out - but aliases was the thing I felt it provided that we don't cover through `IFunction` and it provides a degree of control here.

Initially I wasn't quite familiar enough that versions had their own unique ARN. I'd created a function, published a bunch of versions, and imported it into a sample app (through `fromFunctionArn`). Following that, I assigned aliases to the versions and then using them was where I found I needed this (or some mechanism) to specify and found it useful to have the ability to map an alias.

I'm happy to dig more if needed. As of right now, I'm still converging on the opinion that it provides some value. A mismatch / bad qualifier will fail, but I think for a lot of use cases, leaving it unspecified or working directly with the ARN is smooth.